src/run.c

```diff
@@ -109,7 +109,6 @@ static void run(uint32_t outlen, char *pwd, char *salt, uint32_t t_cost,
         fatal("salt missing");
     }
 
-    pwdlen = strlen(pwd);
```

Comment:
There are a bunch of other uses of `strlen(pwd)` in this function. Now that pwd isn't necessarily null terminated, those should be changed to `pwdlen`.